From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jokubas Maciulaitis <ubitas52@gmail.com>
Date: Fri, 16 Oct 2026 15:25:26 +0300
Subject: [PATCH] net: ti: davinci_emac: wait for the link when the EMAC is
 started

Right after boot the link is often still coming up, and the first
command fails with no link. Wait up to 5 s for it.
---
 drivers/net/ti/davinci_emac.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ti/davinci_emac.c b/drivers/net/ti/davinci_emac.c
index f1e640a9..f6c70076 100644
--- a/drivers/net/ti/davinci_emac.c
+++ b/drivers/net/ti/davinci_emac.c
@@ -522,7 +522,13 @@ static int davinci_emac_start(struct udevice *dev)
 	/* We need to wait for MDIO to start */
 	udelay(1000);
 
-	index = get_active_phy();
+	/* right after boot the link may still be coming up */
+	for (cnt = 0; cnt < 50; cnt++) {
+		index = get_active_phy();
+		if (index != -1)
+			break;
+		mdelay(100);
+	}
 	if (index == -1)
 		return(0);
 
